spec/features/metrics_spec.rb

```diff
@@ -0,0 +1,222 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'metrics tracking', :js do
+  before do
+    allow(Settings).to receive(:metrics_api_url).and_return('/test_metrics_api')
+    allow(Embed::Purl).to receive(:find).and_return(purl)
+    StubMetricsApi.reset!
+  end
+
+  after do
+    FileUtils.rm_rf('tmp/test_downloads')
+  end
```

Comment:
without this, running the tests will occasionally download an actual file, because it happens extremely fast. the new capybara config ensures it goes into `tmp/` and then we delete it afterwards